### ISSUE TITLE
URLEncoder.encode(String, String) false positive

### ISSUE THREAD

**Issue by rdehuyss** [google/error-prone#5558]
We're using the following code in our project:

```java
return URLEncoder.encode(string, UTF_8.name());
````

When using errorprone with gradle we get a warning message that is obsolete. This is [not the case](https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/net/URLEncoder.html#encode(java.lang.String,java.lang.String)) however.

### Build output
```
     /jobrunr/core/src/main/java/org/jobrunr/utils/StringUtils.java:112: warning: [JdkObsolete] Use URLEncoder.encode(String, Charset) instead.
            return URLEncoder.encode(string, UTF_8.name());
                                    ^
    (see https://errorprone.info/bugpattern/JdkObsolete)
```

### Our setup:
```gradle
plugins {
    id 'net.ltgt.errorprone' version '5.1.0' apply false
}

dependencies {
   errorprone 'com.google.errorprone:error_prone_core:2.48.0'
}
```

**Comment by cushon** [comment]
The recommendation is to use `URLEncoder.encode(string, UTF_8)`, instead of passing the `Charset` as a `String`

**Comment by rdehuyss** [comment]
> The recommendation is to use `URLEncoder.encode(string, UTF_8)`, instead of passing the `Charset` as a `String`

I know but this is not possible in JDK 8.

**Comment by mirkoalicastro** [comment]
Hi @cushon! Same here. I've addressed this with a PR that naturally extends the existing `androidMinSdkVersion` approach with an analogous `jdkVersion` field. Could you pls have a look? https://github.com/google/error-prone/pull/5648
